### DISCUSSION TITLE
Anybody had success implementing Highcharts?

### DISCUSSION THREAD

**Discussion by viniciusbo** [remix-run/remix#8153] [Q&A] [answered]
Hello there. First time trying out Remix for a new project.

I'm trying to add Highcharts to my Remix project but I can't get it working. The following error is thrown:

`Error: Element type is invalid: expected a string (for built-in components) or a class/function (for composite components) but got: object.`

I think the issue could be due to Remix trying to render this code on the server, but even forcing client-side rendering/bundling, the app returns the error above.

Does anybody had success implementing Highcharts in Remix?

I've also tried using `chart.client.jsx` along with `ClientOnly` for the Chart component, but no success. Also tried rendering everything on the server, but also couldn't get it to work. Also tried searching stackoverflow/github for any previous attempt to make Highcharts work with Remix, but couldn't find anything.

What am I doing wrong? Any advices towards the right path? Thank you in advance.

Here's the code:

root.jsx
 ```root.jsx
import {
  Links,
  Meta,
  Outlet,
  Scripts,
} from "@remix-run/react";
import Chart from "../chart";

export default function App() {
  return (
    <html>
      <head>
        <link
          rel="icon"
          href="data:image/x-icon;base64,AA"
        />
        <Meta />
        <Links />
      </head>
      <body>
        <h1>Hello world!</h1>

        <Chart />

        <Outlet />
        <Scripts />
      </body>
    </html>
  );
}
```

chart.jsx
```chart.jsx
import { ClientOnly } from "remix-utils/client-only";
import Highcharts from 'highcharts'
import HighchartsReact from 'highcharts-react-official'

const options = {
  title: {
    text: 'My chart'
  },
  series: [{
    data: [1, 2, 3]
  }]
}

export default function Chart() {
  return (
    <ClientOnly fallback={'Fallback'}>
      {() => (
        <div>
          <HighchartsReact
            highcharts={Highcharts}
            options={options}
          />
        </div>
      )}
    </ClientOnly>
  )
}
```

remix.config.cjs
```js
module.exports = {
  serverDependenciesToBundle: [
    /^remix-utils.*/,
    "highcharts",
  ],
};

```

**Comment by kiliman** [comment]
Here's a StackBlitz

https://stackblitz.com/edit/remix-run-remix-xnlpw8?file=app%2Froutes%2F_index.tsx

**Reply by kiliman** [reply, marked as answer]
<img width="700" alt="image" src="https://github.com/remix-run/remix/assets/47168/9080ae2c-225a-41c2-abf6-c81e335e1746">

**Reply by viniciusbo** [reply]
I really appreciate the time you took to make it work. Now I know how this situation should be handled. Thank you.

**Reply by kiliman** [reply]
Yes, typically, when dealing with components that don't support SSR, I simply re-export everything from a _*.client_ file and import from there.

**Comment by khanhnqnd** [comment]
**Import**
```tsx
import HighchartsReact from "highcharts-react-official";
import Highcharts from 'highcharts';
```

**Use CSR (Client Side Rendering)**
```tsx
 const [isClient, setIsClient] = useState(false);
  useEffect(() => {
    setIsClient(true);
  }, []);
```

**Return function**
```tsx
 return (
    <>
      {
        isClient ? (
          <HighchartsReact highcharts={Highcharts}/>
        ) : null
      }
    </>
  )
```
